contributors/design-proposals/flakiness-sla.md

```diff
@@ -0,0 +1,92 @@
+are out of the scope of the document, but some currently known challenges
+are listed at the end of the document.
+
+## Flakiness SLA
+We will measure flakiness based on pull requests that are run through pre-submit
+PR testing. The metric that we will monitor is: Pre-submit flakes per PR. This metric
+will be calculated on a daily and weekly basis using:
+
+Sum(Flakes not caused by the PR) / Total(PRs)
+
+Our current SLA is that this metric will be less than 0.01
+(1% of all PRs have flakes that are not caused by the PR itself).
+
+## Activities on SLA violation
+When the project is out of SLA for flakiness on either the daily or weekly metric
+we will stop the merge queue.
+
+Only PRs with the label `queue/fix` or `queue/critical-fix` will be allowed to merge.
+The merge queue will remain blocked until a sufficient number of flake-fixes have been
+submitted to restore the daily CLA. We will measure this each day by via automation that
+re-tests a number of PRs and looking for more flakes.
+
+### Exceptions
+We may consider merging PRs under exceptional circumstances, for example
+merging security fixes or other critical bug fixes that need to be cut into
+a pre-existing release. Each exception will be considered on a case by case
+basis. Exceptions will only be granted for PRs that are already labelled with
+cherry-pick approve.
+
+## Non-goals: Flakiness at HEAD
+We will consider flakiness for PRs only, we will _not_ currently 
+measure flakiness for continuous e2e that are run at HEAD independent of PRs.
+
+There are a number of reasons for this:
+   * We already block the submit-queue if certain HEAD e2e tests are failing
+   * The rate at which code is merged at HEAD is so great that it's rare that a test will be re-run for the exact same git commit.
+   * The volume of testing on PRs is significantly higher than at HEAD, so flakes are more readily apparent.
+   * The flakiness of these tests is already being measured in other places.
+
+## Approaches to measuring the flakiness metric
+
+Currently, measuring this metric is somewhat tricky, since determining flakes
+caused by PRs vs. existing flakes is somewhat challenging. We will use a variety
+of heuristics to determine this, including looking at PRs that contain changes
+that could not cause flakes (e.g. typos, docs). As well as looking at the
+past history of test failures (e.g. if a test fails across many different PRs
+it's likely a flake)
+
+### Detecting changes in the environment (e.g. GCE, Azure, AWS, etc)
+Changes in the environment are notably hard to measure or control for, but we'll do our best.
+
+### Detecting changes in code
+Changes in code are obviously quite easy to control. The github API can tell us if the commit SHA
+has changed between different runs of the test result.
+
+### Detecting if it is caused by PR or pre-existing?
+Measuring whether a flake is caused by a PR or existing problems is a challenge, but will use
```

Comment:
Not only is it a "challenge" to tell flaky/not-flaky, it's equally important to tell when a PR is changing the rate of a given test from 1/1000 to 2/1000. (has same effect on SLA)